### ISSUE TITLE
Create `/auth/cli/download`

### ISSUE THREAD

**Issue by meain** [paralus/paralus#4]
Fetch latest release from Github API and provide to the frontend for direct download.

**Comment by meain** [comment]
- What version should it download? `latest` or should we have some way to encode the compatible version which ideally the latest one should be.
- Will GitHub ratelimit api requests? Should we cache the responses? Any possible fallbacks like hard coding urls for download.

**Comment by akshay196** [comment]
> What version should it download? latest or should we have some way to encode the compatible version which ideally the latest one should be.

Answer should come from what is the compatibility relation between CLI and Paralus. We currently not decided this relation yet. It's good time to think and come up with compatibility matrix. The latest version of CLI may or may not always compatible with the Paralus version someone is using.
Created separate ticket to decide on compatibility matrix: https://github.com/paralus/paralus/issues/214

> Will GitHub ratelimit api requests? Should we cache the responses? Any possible fallbacks like hard coding urls for download.

Currently for unauthenticated user, the GitHub sets rate limit to 60 requests per hour which is fair number for getting-started user or low to mid-size group of users. We may reduce the GitHub calls by caching the binary at back-end or front-end which can be done using [ETag or Last-Modified headers](https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#conditional-requests).
However for really big organization Paralus should have option to set GitHub token that would increase the rate limit to 5000 requests per hour.

Few other topics to discuss before we start implementation:
1. Is the Dashboard only client which requires this endpoint? If yes, should we move this complete download facility to Dashboard repo?
2. Should we provide download for different architectures (x86, x86_64, arm) along with OS and distros?
3. Will this CLI download API placed behind authentication and authorization?

**Comment by akshay196** [comment]
Closing as we no indent to support this feature.
Reopen if you're interested of having this feature.
